pkg/compactv2/modifiers.go

```diff
@@ -326,4 +330,103 @@ func (p *delChunkSeriesIterator) Next() bool {
 
 func (p *delChunkSeriesIterator) At() chunks.Meta { return p.curr }
 
-// TODO(bwplotka): Add relabelling.
+type RelabelModifier struct {
+	relabels []*relabel.Config
+}
+
+func WithRelabelModifier(relabels ...*relabel.Config) *RelabelModifier {
+	return &RelabelModifier{relabels: relabels}
+}
+
+func (d *RelabelModifier) Modify(_ index.StringIter, set storage.ChunkSeriesSet, log ChangeLogger, p ProgressLogger) (index.StringIter, storage.ChunkSeriesSet) {
```

Comment:
I feel there are some bugs in my Modifier implementation. But Idk why.
The weird thing is that after changing the reference in pr https://github.com/thanos-io/thanos/pull/3710, it works. But ideally the existing `deletion modifier` works even without that pr.

Reply:
Solved. The root cause is the lazy population of the chunk and buf chunks reuse.